ITISFoundation/osparc-simcore
# New classifiers with scicrunch RRIDs (⚠️ devops)

## What do these changes do?

This PR extends classifiers to include using K-Core's curated list of research resource identifiers (RRIDs) from scicrunch's [RRID Portal](https://scicrunch.org/resources). 

### what are classifiers?
- Every group has a set of "official" classifiers and its users tag studies with them
- Classifiers can be defined in two ways:
    1.  a static tree (or *bundle*) that is stored in ``group_classifiers.c.bundle``
    2. using RRIDs from scicrunch.org (see ``group_classifiers.c.uses_scicrunch`` )
- In the UI, classifiers are show in a tree that represents
    1. one-to-one the bundle tree
    2. a dynamic tree built from valid scicrunch RRIDs (currently always flat list)

### Design
This has been implemented in the ``groups`` apps module of the ``webserver`` that uses a new submodule ``scicrunch`` to interact with scicrunch.org.

<img width="1104" alt="Screen Shot 2020-12-17 at 11 41 05" src="https://user-images.githubusercontent.com/32402063/102477528-c1f28f00-405c-11eb-9d9f-7dacf852157b.png">




## Related issue/s

- ITISFoundation/osparc-issues#344: J-D4.6. A computational Minimal Information Standard (cMIS) has been established Y4M03


## How to test



### manual testing

Current PR is deployed in my machine (see link in Mattermost) with the following setup 

- Test user (login: user@itis.swiss, pwd : test) belongs to NIH-sparc group
- "NIH-sparc" group uses scicrunch classifiers (i.e. RRID-based classifiers)
- "Everyone" group has bundle classifiers 
- If a user belongs to several groups all classifiers are shown together

Possible checks 
- classifiers show and info display. Links to scicrunch.org RRIDs are correct
- can add new RRID to classifiers
- can assign a classifier to a study
- ...

### unit/integration tests


```cmd
make devenv
cd services/web/server
make install-dev
make tests
```


## Deployment

- ⚠️ REQUIRED: setup new environment variables in ``.env``:   
```
SCICRUNCH_API_BASE_URL=https://scicrunch.org/api/1
SCICRUNCH_API_KEY=REPLACE_ME_with_valid_api_key
```
For API_KEY, create account and get one key for each deploy here  [here](https://scicrunch.org/account/developer))



## Checklist

Next PR.
~- [ ] FIX: tests~
~- [ ] ValidationErrors to 422 Unprocessable Entity~ 
~- [ ] Unit tests for the changes exist~
